### ISSUE TITLE
1002: Verification Error: Execution: Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress: RuntimeApi, Execution: Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress

### ISSUE THREAD

**Issue by HamzaYasin1** [centrifuge/go-substrate-rpc-client#289]
`1002: Verification Error: Execution: Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress: RuntimeApi, Execution: Could not convert parameter `tx` between node and runtime: No such variant in enum MultiAddress`

This issue wasn't in substrate 2.0.0 but after upgrading the runtime to 2.0.1 so facing this error.

You can produce this issue while using the below repo. https://github.com/substrate-developer-hub/substrate-node-template/tree/v2.0.1

If anyone knows what are the types for this repo, please let me know.

Thanks

**Comment by cdamian** [comment]
Is this still something that we can help with? Can you also provide more details about when this error was encountered? Were you trying to execute an extrinsic?

**Comment by HamzaYasin1** [comment]
This problem arises when there is a mismatch in developer types. However, when we correctly match the types, it functions properly.

I'm still eager to understand the precise method for calculating the required types for my chain, especially when incorporating a custom pallet.

**Comment by cdamian** [comment]
Yes, if you are using a non-default `MultiAddress` on your chain then you would need to update it accordingly. The same goes with most of the types that are using when interacting using this lib, either when retrieving events or reading storages.
